### PR TITLE
:sparkles: 1208 Handle lunch lectures through the webshop

### LINKED ISSUE #1208: ✨ Handle lunch lectures through the shop
- [ ] Food preferences can be set by members
- [ ] Tickets (products) can be created through the frontend
- [ ] List of attendees (with food preferences) can be exported

### PR DESCRIPTION
**Pull request by oliverlevay**
Closes #1208

### CODE REVIEW

File: frontend/public/locales/en/common.json [hunk 1]
```diff
@@ -104,5 +104,9 @@
   "checklist": "Checklist",
   "packinglist": "Packing list",
   "studenthealth": "Student Health",
-  "schedule": "Schedule"
+  "schedule": "Schedule",
```
Comment (danieladugyan) [inline frontend/public/locales/en/common.json:107]: not the same as the swedish translation?

File: frontend/providers/UserProvider.tsx [hunk 2]
```diff
@@ -60,6 +61,11 @@ export function UserProvider({ children }: PropsWithChildren<{}>) {
 
   return (
     <userContext.Provider value={memoized}>
+      <AddFoodPreferencePopup
+        id={user?.id}
+        open={(user?.student_id && user?.food_preference === null)}
+        refetchUser={refetch}
+      />
```
Comment (danieladugyan) [inline frontend/providers/UserProvider.tsx:68]: Does this need to be in the UserProvider? (SoC)

File: frontend/providers/AddFoodPreferencePopup.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        <Button
+          color="primary"
+          onClick={() => updateFoodPreferenceMutation({
+            variables: {
+              id,
+              foodPreference: '',
+            },
+          }).then(() => refetchUser())}
+        >
+          Jag har ingen
```
Comment (danieladugyan) [inline frontend/providers/AddFoodPreferencePopup.tsx:53]: translation?

File: frontend/providers/AddFoodPreferencePopup.tsx [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+          color="primary"
+          onClick={() => updateFoodPreferenceMutation({
+            variables: {
+              id,
+              foodPreference,
+            },
+          }).then(() => refetchUser())}
+          disabled={!foodPreference}
+        >
+          Save
```
Comment (danieladugyan) [inline frontend/providers/AddFoodPreferencePopup.tsx:65]: translation?

File: frontend/components/Webshop/Product.tsx [hunk 5]
```diff
@@ -125,26 +166,46 @@ export default function Product({ product }: { product: ProductsQuery['products'
...
+            >
+              <AddShoppingCartIcon style={{ marginRight: '1rem' }} />
+              {t('webshop:add_to_cart')}
+            </Button>
+            )
+          }
+          {
+            timeLeft > 0 && (
+              <Typography>
+                Biljetter släpps om:
```
Comment (danieladugyan) [inline frontend/components/Webshop/Product.tsx:203]: more translations?

File: frontend/components/Webshop/Product.tsx [hunk 6]
```diff
@@ -30,10 +32,31 @@ function getQuantityInMyCart(productId: string, myCart?: MyCartQuery['myCart'])
...
+  const hours = Math.floor((duration / (1000 * 60 * 60)) % 24);
+  const days = Math.floor((duration / (1000 * 60 * 60 * 24)));
+  let output = '';
+  if (duration > (1000 * 60 * 60 * 24)) { output += `${days}d `; }
+  if (duration > (1000 * 60 * 60)) { output += `${hours}h `; }
+  if (duration > (1000 * 60)) { output += `${minutes}m `; }
+  output += `${seconds}s`;
+  return output;
+};
+
```
Comment (danieladugyan) [inline frontend/components/Webshop/Product.tsx:54]: Can't we use Luxon to do all of this?

**Comment by danieladugyan** [comment]
Also, lots of nolla changes that shouldn't be here. Probably just caused by @Macludde force push.

File: frontend/components/Webshop/Product.tsx (hunk 6 above)
Comment (oliverlevay) [reply frontend/components/Webshop/Product.tsx:54]: I couldn't find a nice solution

**Pull request merged**